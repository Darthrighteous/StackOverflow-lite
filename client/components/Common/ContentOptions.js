import React from 'react';
import PropTypes from 'prop-types';

const ContentOptions = (props) => {
  const {
    title, sort, dropdownOpenState, onDropdownClick, onSortClick, background
  } = props;
  return (
    <div className="content-options" style={(background) ? { background: '#f5f5f5' } : null}>
      <h4>{title}</h4>
      <div role="button" tabIndex="0" onKeyDown={null} className="dropdown" onClick={onDropdownClick}>
        <span className="dropdown-title">Sort</span>
        <div className="dropdown-container">
          <button type="button" className="drop-btn">{sort}</button>
          <svg className="drop-icon" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg">
            <g>
              <path fill="inherit" d="M14.1711599,9.3535 L9.99925636,13.529 L5.82735283,9.3535 C5.51262415,9.0385 5.73543207,8.5 6.18054835,8.5 L13.8179644,8.5 C14.2630807,8.5 14.4858886,9.0385 14.1711599,9.3535" />
            </g>
          </svg>
        </div>

        {(dropdownOpenState) ? (
          <div className="dropdown-content">
            <button
              type="button"
              className="dropdown-item active"
              onClick={onSortClick.bind(this, 'most recent')}
            >
              Most Recent
            </button>

            <button
              type="button"
              className="dropdown-item"
              onClick={onSortClick.bind(this, 'most answered')}
            >
              Most Answered
            </button>

            <button
              type="button"
              className="dropdown-item"
              onClick={onSortClick.bind(this, 'most rated')}
            >
              Most Rated
            </button>
          </div>
        ) : (
          <div />
        )}
      </div>
    </div>);
};

ContentOptions.propTypes = {
  sort: PropTypes.string.isRequired,
  title: PropTypes.string,
  background: PropTypes.bool,
  dropdownOpenState: PropTypes.bool.isRequired,
  onDropdownClick: PropTypes.func.isRequired,
  onSortClick: PropTypes.func.isRequired
};

ContentOptions.defaultProps = {
  title: '',
  background: false,
};

export default ContentOptions;