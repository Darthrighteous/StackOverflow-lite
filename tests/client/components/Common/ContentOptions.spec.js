import React from 'react';
import { shallow } from 'enzyme';

import ContentOptions from '../../../../client/components/Common/ContentOptions';

const props = {
  sort: 'most recent',
  dropdownOpenState: true,
  onDropdownClick: jest.fn(),
  onSortClick: jest.fn(),
};

const bgProps = {
  ...props,
  background: true,
};

describe ('content options test', () => {
  const contentOptionsWrapper = shallow(
    <ContentOptions {...props} />
  );

  const withBackground = shallow(<ContentOptions {...bgProps} />);

  test('snapshot tests', () => {
    expect(contentOptionsWrapper).toMatchSnapshot();
    expect(withBackground).toMatchSnapshot();
  });

  test('event simulations', () => {
    contentOptionsWrapper.find('.dropdown').simulate('click');
    expect(contentOptionsWrapper.state().isDropdownOpen).toBe(true);

    const instance = contentOptionsWrapper.instance();
    instance.closeDropdown();
    expect(contentOptionsWrapper.state().isDropdownOpen).toBe(false);
  });
});
