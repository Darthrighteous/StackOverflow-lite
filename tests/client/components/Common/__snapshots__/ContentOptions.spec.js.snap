// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content options test snapshot tests 1`] = `
<div
  className="content-options"
  style={null}
>
  <h4 />
  <div
    className="dropdown"
    onClick={[MockFunction]}
    onKeyDown={null}
    role="button"
    tabIndex="0"
  >
    <span
      className="dropdown-title"
    >
      Sort
    </span>
    <div
      className="dropdown-container"
    >
      <button
        className="drop-btn"
        type="button"
      >
        most recent
      </button>
      <svg
        className="drop-icon"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M14.1711599,9.3535 L9.99925636,13.529 L5.82735283,9.3535 C5.51262415,9.0385 5.73543207,8.5 6.18054835,8.5 L13.8179644,8.5 C14.2630807,8.5 14.4858886,9.0385 14.1711599,9.3535"
            fill="inherit"
          />
        </g>
      </svg>
    </div>
    <div
      className="dropdown-content"
    >
      <button
        className="dropdown-item active"
        onClick={[Function]}
        type="button"
      >
        Most Recent
      </button>
      <button
        className="dropdown-item"
        onClick={[Function]}
        type="button"
      >
        Most Answered
      </button>
      <button
        className="dropdown-item"
        onClick={[Function]}
        type="button"
      >
        Most Rated
      </button>
    </div>
  </div>
</div>
`;

exports[`content options test snapshot tests 2`] = `
<div
  className="content-options"
  style={
    Object {
      "background": "#f5f5f5",
    }
  }
>
  <h4 />
  <div
    className="dropdown"
    onClick={[MockFunction]}
    onKeyDown={null}
    role="button"
    tabIndex="0"
  >
    <span
      className="dropdown-title"
    >
      Sort
    </span>
    <div
      className="dropdown-container"
    >
      <button
        className="drop-btn"
        type="button"
      >
        most recent
      </button>
      <svg
        className="drop-icon"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M14.1711599,9.3535 L9.99925636,13.529 L5.82735283,9.3535 C5.51262415,9.0385 5.73543207,8.5 6.18054835,8.5 L13.8179644,8.5 C14.2630807,8.5 14.4858886,9.0385 14.1711599,9.3535"
            fill="inherit"
          />
        </g>
      </svg>
    </div>
    <div />
  </div>
</div>
`;
